plugin/session_view.py

```diff
@@ -127,6 +127,12 @@ def present_diagnostics_async(self, flags: int) -> None:
                 self.view.erase_regions(key)
             elif ((severity <= userprefs().show_diagnostics_severity_level) and
                     (data.icon or flags != (sublime.DRAW_NO_FILL | sublime.DRAW_NO_OUTLINE))):
+                def handle_same_regions(region: sublime.Region):
+                    # this allows showing diagnostics with same begin and end in the view
+                    if region.begin() == region.end():
+                        return sublime.Region(region.begin(), region.begin() + 1)
+                    return region
+                data.regions = list(map(handle_same_regions, data.regions))
```

Comment:
I would rather see this check be done in `SessionBuffer.on_diagnostics_async`, because this "present" method is now also mutating the diagnostics data which can be confusing. See line 256 of session_buffer.py.

Reply:
Instead of mutating the data. I could just create a new variable:
```diff
-data.regions = list(map(handle_same_regions, data.regions))
+underline_regions = list(map(handle_same_regions, data.regions))

-self.view.add_regions(key, data.regions, data.scope, data.icon, flags)
+self.view.add_regions(key, underline_regions, data.scope, data.icon, flags)
```


I would leave it here, because I only wanted to fix the underlines. I don't want to modify the response from the server. 

What do you think about that,
or do you still think that it is better to put this check in `on_diagnostics_async`?

Reply:
I agree, good point